pkg/crc/preflight/preflight.go

```diff
@@ -103,6 +105,18 @@ func (check *Check) doCleanUp() error {
 	return check.cleanup()
 }
 
+func sort(checks []Check) {
+	gosort.Slice(checks, func(i, j int) bool {
+		if checks[i].order < checks[j].order {
+			return true
+		}
+		if checks[i].order > checks[j].order {
+			return false
+		}
+		return checks[i].configKeySuffix < checks[j].configKeySuffix
+	})
+}
```

Comment:
Looking at the comparison function, this is ever more complicated than I thought, the genericPreflightChecks won't be executed in the order they are listed in the source file even if they have the same order, if one forgets to set `order`, their check will be the very first one, ... Overall I fear it's going to be very hard to predict the order the tests are going to be run in without looking at the code, when you add a new test, you have to check the order of most of the other ones if you want it to be run at the appropriate time, ... I'd really favour avoiding this field, and just tagging the few tests that we want to be first or last, but for all the others, just use the order they are in the `getPreflightChecks` call (before sorting the array).

Reply:
yes, not easy. I found it also not really nice to read.

I would like to do the same as init or OpenShift to organize their manifests (0000_50_foo.yaml, etc.) Is there a way to do that more nicely?

Reply:
I never found that way of ordering very nice when you are not very familiar with the project. When you want to add a new file, which priority should you pick? How do you know if it should be run early/late, if some priorities are reserved or not, if you can reuse the same priority as another file or not, ... 

I still think that just a RunFirst/RunLast tag for the few tests we want to (de)prioritize would be plenty enough, and easy to understand. I think it would work ok for us as we won't have hundreds of preflights.